modules/sugar/src/main/ts/ephox/sugar/selection/query/Within.ts

```diff
@@ -10,8 +10,8 @@ import * as SelectionDirection from '../core/SelectionDirection';
 
 const withinContainer = <T extends Element>(win: Window, ancestor: SugarElement<Element>, outerRange: Range, selector: string): SugarElement<T>[] => {
   const innerRange = NativeRange.create(win);
-  const self = Selectors.is<T>(ancestor, selector) ? [ ancestor ] : [];
-  const elements = self.concat(SelectorFilter.descendants<T>(ancestor, selector));
+  const ancestorArray = Selectors.is<T>(ancestor, selector) ? [ ancestor ] : [];
```

Comment:
Why is this needed to change is there some new rule that prevents you from creating local variables that shadow globals. If so then there will be a lot of issues with this sine `name` for example is a global.

Reply:
Oh seems to be the `consistent-this` rule. I guess that one would make sense to have.